lib/merge.js

```diff
@@ -6,9 +6,13 @@ function merge (original, rels) {
     , rel = rels[id]
   if (rel) {
     rels[id] = null
-    var doc = _.defaults.apply(null, [original].concat(rel))
-    doc.sources = _.compact(_.flatten(_.pluck(rel, 'sources')))
-    return [ doc ]
+    if (rel.length) {
+      var doc = _.defaults.apply(null, [original].concat(rel))
+      doc.sources = _.compact(_.flatten(_.pluck(rel, 'sources')))
```

Comment:
just for the future, in similar cases you can probably try to use [chaining](http://devdocs.io/lodash/index#prototype-chain), even if it can be a bit nasty and i failed often using it. worth a try if your tests are covering you! :)